providers/base/bin/snap_confinement_test.py

```
@@ -17,64 +17,173 @@
+        exit_code = 0
+        for snap in data:
+            snap_name = snap.get("name")
+            snap_confinement = snap.get("confinement")
+            snap_devmode = snap.get("devmode")
+            snap_revision = snap.get("revision")
+
+            if snap_name is None:
+                logging.error("Snap 'name' not found in the snap data.")
+                exit_code = 1
+
+            if any(
+                re.match(pattern, snap_name)
+                for pattern in self.whitelist_snaps
+            ):
+                print("Skipping whitelisted snap: {}".format(snap_name))
+                continue
+
+            if snap_confinement != "strict":
+                exit_code = 1
+                logging.error(
+                    "Snap '%s' confinement is expected to be 'strict' "
+                    "but got '%s'", snap_name, snap_confinement,
+                )
+
+            if snap_devmode is not False:
+                exit_code = 1
+                logging.error(
+                    "Snap '%s' devmode is expected to be False but "
+                    "got '%s'", snap_name, snap_devmode,
+                )
+
+            if snap_revision and snap_revision.startswith("x"):
+                exit_code = 1
+                logging.error(
+                    "Snap '%s' has sideloaded revision '%s', which "
+                    "is not allowed", snap_name, snap_revision,
+                )
+            elif snap_revision is None:
+                exit_code = 1
+                logging.error(
+                    "'revision' not found in snap '%s'", snap_name,
+                )
+        return exit_code
+
+
+def main():
+    logging.basicConfig(format='%(levelname)s: %(message)s')
+    sub_commands = {
+        "system": SystemConfinement,
+        "snaps": SnapsConfinement,
+    }
+    parser = argparse.ArgumentParser()
+    parser.add_argument("subcommand", type=str, choices=sub_commands)
+    args = parser.parse_args(sys.argv[1:2])
```

Any specific reason for parsing only one string here?
I'd guess that doing a general `parser.parse_args()` would achieve the same in normal cases **plus** it would complain when someone would provide some extra things. Those extra things would be silently ignored with the `[1:2]` approach.

---

Indeed, I'll apply what you suggest in the next commit :smile: